lib/phoenix_library/books.ex

```diff
@@ -0,0 +1,202 @@
+         })
+
+    {:ok, %PhoenixLibrary.Books.Book{...}}
+
+    iex> PhoenixLibrary.Books.create_book(%{
+           title: "Adopting Elixir"
+         })
+
+    {:error, #Ecto.Changeset<..., valid?: false>}
+  """
+  def create_book(params) do
+    params
+    |> Book.create_changeset()
+    |> Repo.insert()
+  end
+
+  @doc """
+  Retorna uma tupla {:ok, <livro>} caso o livro seja encontrado no banco de dados
+  ou uma tupla {:erro, :not_found} caso o livro não possa ser encontrado.
+
+  ## Parâmetros
+
+    - uuid: id do livro a ser buscado no formato UUID.
+
+  ## Exemplos
+
+    iex> PhoenixLibrary.Books.load_book("0200e8de-69c0-4d88-982c-31721cb4af5c")
+
+    {:ok, %PhoenixLibrary.Books.Book{...}}
+
+    iex> PhoenixLibrary.Books.load_book("521675b8-df5d-495d-b579-4f2c69716318")
+
+    {:error, #Ecto.Changeset<..., valid?: false>}
+  """
+  def load_book(uuid) do
+    case Repo.get(Book, uuid) do
+      nil -> {:error, :not_found}
+      data -> {:ok, data}
+    end
+  end
+
+  @doc """
+  Retorna uma tupla {:ok, <livro>} caso o livro seja encontrado no banco de dados
+  ou uma tupla {:erro, :not_found} caso o livro não possa ser encontrado.
+
+  ## Parâmetros
+
+    - params: Map com parametros a serem atualizados no livro. Deve conter uma chave
+      "id" com o id do livro a ser atualizado.
+
+  ## Exemplos
+
+    iex> PhoenixLibrary.Books.update_book(%{
+           "id" => "0200e8de-69c0-4d88-982c-31721cb4af5c",
+           "title" => "Adopting Elixir - From Concept to Production"
+         })
+
+    {:ok, %PhoenixLibrary.Books.Book{...}}
+
+    iex> PhoenixLibrary.Books.update_book(%{
+           "id" => "521675b8-df5d-495d-b579-4f2c69716318",
+           "title" => "Adopting Elixir - From Concept to Production"
+         })
+
+    {:error, :not_found}
+  """
+  def update_book(%{"id" => id} = params) do
+    case load_book(id) do
+      {:error, :not_found} -> {:error, :not_found}
+      {:ok, entity} -> update_entity(entity, params)
+    end
+  end
+
+  @doc """
+  Retorna uma tupla {:ok, <livro>} caso o livro seja encontrado no banco de dados
+  ou uma tupla {:erro, :not_found} caso o livro não possa ser encontrado.
+
+  ## Parâmetros
+
+    - id: id do livro a ser buscado no formato UUID.
+
+  ## Exemplos
+
+    iex> PhoenixLibrary.Books.delete_book("0200e8de-69c0-4d88-982c-31721cb4af5c")
+
+    {:ok, %PhoenixLibrary.Books.Book{...}}
+
+    iex> PhoenixLibrary.Books.update_book("521675b8-df5d-495d-b579-4f2c69716318")
+
+    {:error, :not_found}
+  """
+  def delete_book(id) do
+    case load_book(id) do
+      {:error, :not_found} -> {:error, :not_found}
+      {:ok, book} -> Repo.delete(book)
+    end
+  end
+
+  @doc """
+  Retorna uma lista com todos os livros cadastrados no banco de dados.
+
+  ## Exemplos
+
+    iex> PhoenixLibrary.Books.list_books()
+
+    [%PhoenixLibrary.Books.Book{...}, %PhoenixLibrary.Books.Book{...}]
+  """
+  def list_books do
+    Book
+    |> Repo.all()
```

Comment:
nit, se apenas um pipe é utilizado, é interessante chamar direto, mas isso é opcional
```suggestion
    Repo.all(Book)
```